proto/cosmwasm/wasm/v1/query.proto

```diff
@@ -17,65 +18,77 @@ service Query {
+    option (cosmos.query.v1.module_query_safe) = true;
     option (google.api.http).get =
         "/cosmwasm/wasm/v1/contract/{address}/history";
   }
   // ContractsByCode lists all smart contracts for a code id
   rpc ContractsByCode(QueryContractsByCodeRequest)
       returns (QueryContractsByCodeResponse) {
+    option (cosmos.query.v1.module_query_safe) = true;
     option (google.api.http).get = "/cosmwasm/wasm/v1/code/{code_id}/contracts";
   }
   // AllContractState gets all raw store data for a single contract
   rpc AllContractState(QueryAllContractStateRequest)
       returns (QueryAllContractStateResponse) {
+    option (cosmos.query.v1.module_query_safe) = true;
     option (google.api.http).get = "/cosmwasm/wasm/v1/contract/{address}/state";
   }
   // RawContractState gets single key from the raw store data of a contract
   rpc RawContractState(QueryRawContractStateRequest)
       returns (QueryRawContractStateResponse) {
+    option (cosmos.query.v1.module_query_safe) = true;
     option (google.api.http).get =
         "/cosmwasm/wasm/v1/contract/{address}/raw/{query_data}";
   }
   // SmartContractState get smart query result from the contract
   rpc SmartContractState(QuerySmartContractStateRequest)
       returns (QuerySmartContractStateResponse) {
+    option (cosmos.query.v1.module_query_safe) = true;
     option (google.api.http).get =
         "/cosmwasm/wasm/v1/contract/{address}/smart/{query_data}";
   }
   // Code gets the binary code and metadata for a singe wasm code
   rpc Code(QueryCodeRequest) returns (QueryCodeResponse) {
+    option (cosmos.query.v1.module_query_safe) = true;
     option (google.api.http).get = "/cosmwasm/wasm/v1/code/{code_id}";
   }
   // Codes gets the metadata for all stored wasm codes
   rpc Codes(QueryCodesRequest) returns (QueryCodesResponse) {
+    option (cosmos.query.v1.module_query_safe) = true;
     option (google.api.http).get = "/cosmwasm/wasm/v1/code";
   }
 
   // PinnedCodes gets the pinned code ids
   rpc PinnedCodes(QueryPinnedCodesRequest) returns (QueryPinnedCodesResponse) {
+    option (cosmos.query.v1.module_query_safe) = true;
     option (google.api.http).get = "/cosmwasm/wasm/v1/codes/pinned";
   }
 
   // Params gets the module params
   rpc Params(QueryParamsRequest) returns (QueryParamsResponse) {
+    option (cosmos.query.v1.module_query_safe) = true;
     option (google.api.http).get = "/cosmwasm/wasm/v1/codes/params";
   }
 
   // ContractsByCreator gets the contracts by creator
   rpc ContractsByCreator(QueryContractsByCreatorRequest)
       returns (QueryContractsByCreatorResponse) {
+    option (cosmos.query.v1.module_query_safe) = true;
     option (google.api.http).get =
         "/cosmwasm/wasm/v1/contracts/creator/{creator_address}";
   }
 
   // BuildAddress builds a contract address
   rpc BuildAddress(QueryBuildAddressRequest)
       returns (QueryBuildAddressResponse) {
+    option (cosmos.query.v1.module_query_safe) = true;
```

Comment:
This query doesn't charge any gas. The option docs say:
> 2. has its gas tracked, to avoid the attack vector where no gas is accounted for on potentially high-computation queries.

I guess this query is so computationally cheap that it's not an issue, but would be good to check somehow.

Reply:
good 👀 ! let's check what @webmaster128  thinks about this

Reply:
This is for creating Instantiate2 addresses, right?

I think this query should be removed as the algorithm is stateless and fixed. There are multiple implementations in Rust, TypeScript and Go. I don't see any reason to perform a query to the node for this.

Reply:
@webmaster128 should we remove this as part of next release? 
I can create a separate pr for it

Reply:
Let's just create a ticket as this might open a can of worms looking at the code. IMO this is not relevant enough to delay the release. We should get CosmWasm 2.1 and other fixes in wasmd out as soon as possible.

Reply:
Would't it be easy to charge a small amount of gas here for the sake of consistency?

Reply:
yes, I think it makes sense to charge some gas for this. I will update the pr